diesel/src/query_dsl/mod.rs

```diff
@@ -797,13 +800,38 @@ pub trait QueryDsl: Sized {
     /// // Executes `SELECT * FROM users FOR UPDATE`
     /// users.for_update().load(&connection)
     /// ```
+    #[cfg(feature = "with-deprecated")]
+    #[allow(deprecated)]
     fn for_update(self) -> ForUpdate<Self>
```

Comment:
Can't `ForUpdateDsl` extend, and be implemented for all, `LockingDsl<ForUpdate>` avoiding the need for two definitions of this method?

Reply:
Hm, I suppose adding a super trait would be a breaking change. If only trait aliases were stable...

Reply:
This would break any existing code that implements `ForUpdateDsl`.